lib/internal/watchdog.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const { JsSigintWatchdog } = internalBinding('watchdog');
+
+class SigintWatchdog extends JsSigintWatchdog {
+  _started = false;
+  _effective = false;
```

Comment:
Properties like this are still very much unoptimized in V8, so it'd be better to set them in a `constructor()`.

Reply:
Having benchmarked on initiating class fields with new class field syntax and constructors, I found that there is no significant performance difference on accessing (getting or writting) the initiated fields but does have not trivial difference on `new`ing instances (0.1ms vs 0.001ms). Yet considering SigintWatchdog doesn't get instantiated continuously, I'd wonder keeping the code easy to read might be more appropriate. What do you think?